commons/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java

```diff
@@ -45,23 +48,54 @@
 
    static {
       // Classes always allowed
+      // Primitives
+      SYS_ALLOWED_CLASSES.add(byte.class.getName());
+      SYS_ALLOWED_CLASSES.add(short.class.getName());
+      SYS_ALLOWED_CLASSES.add(int.class.getName());
+      SYS_ALLOWED_CLASSES.add(long.class.getName());
+      SYS_ALLOWED_CLASSES.add(float.class.getName());
+      SYS_ALLOWED_CLASSES.add(double.class.getName());
+      SYS_ALLOWED_CLASSES.add(char.class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean.class.getName());
+
+      // Primitive Arrays
+      SYS_ALLOWED_CLASSES.add(byte[].class.getName());
+      SYS_ALLOWED_CLASSES.add(short[].class.getName());
+      SYS_ALLOWED_CLASSES.add(int[].class.getName());
+      SYS_ALLOWED_CLASSES.add(long[].class.getName());
+      SYS_ALLOWED_CLASSES.add(float[].class.getName());
+      SYS_ALLOWED_CLASSES.add(double[].class.getName());
+      SYS_ALLOWED_CLASSES.add(char[].class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean[].class.getName());
+
+      // Boxed Primitives
       SYS_ALLOWED_CLASSES.add(Byte.class.getName());
       SYS_ALLOWED_CLASSES.add(Short.class.getName());
       SYS_ALLOWED_CLASSES.add(Integer.class.getName());
       SYS_ALLOWED_CLASSES.add(Long.class.getName());
       SYS_ALLOWED_CLASSES.add(Float.class.getName());
       SYS_ALLOWED_CLASSES.add(Double.class.getName());
-      SYS_ALLOWED_CLASSES.add(Number.class.getName());
       SYS_ALLOWED_CLASSES.add(Character.class.getName());
       SYS_ALLOWED_CLASSES.add(String.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashSet.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashMap.class.getName());
-      SYS_ALLOWED_CLASSES.add(Date.class.getName());
-      SYS_ALLOWED_CLASSES.add(Enum.class.getName());
       SYS_ALLOWED_CLASSES.add(Boolean.class.getName());
+
+      // Java.math
+      SYS_ALLOWED_CLASSES.add(BigInteger.class.getName());
+      SYS_ALLOWED_CLASSES.add(BigDecimal.class.getName());
+
+      // Util
       SYS_ALLOWED_CLASSES.add(ArrayList.class.getName());
-      SYS_ALLOWED_CLASSES.add(Object[].class.getName());
-      SYS_ALLOWED_CLASSES.add("org.infinispan.jboss.marshalling.commons.JBossExternalizerAdapter");
+      SYS_ALLOWED_CLASSES.add(Date.class.getName());
+      SYS_ALLOWED_CLASSES.add(HashMap.class.getName());
+      SYS_ALLOWED_CLASSES.add(HashSet.class.getName());
+      SYS_ALLOWED_CLASSES.add(Instant.class.getName());
+
+      // Misc
+      SYS_ALLOWED_CLASSES.add(Enum.class.getName());
+      SYS_ALLOWED_CLASSES.add(Number.class.getName());
+
+      // Reference array regex
+      SYS_ALLOWED_REGEXP.add("^\\[[a-zA-Z0-9\\.\\$].*\\;$");
```

Comment:
I think the first char after the `[` can only be a capital letter or another `[`:

https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.2

Reply:
@danberindei I think it can still be lower-case, as a class name starting with a lower case value is valid (just bad practice)

Reply:
Updated so that the regex now handles multi-dimension arrays and does not allow `0-9` as the first character.

Reply:
The first character for all classes is `L`, only the primitives get a different (upper-case) letter.